### PR TITLE
added support for nicknamed routes

### LINKED ISSUE #481: Can't override nickname when using namespaces and route params
I have the following API end-point:
```
    class Products < Grape::API
      namespace :products do
        route_param :product_id, type: Integer
          desc 'Return a given product by id.', nickname: 'getProductById'
          get do
          end
        end
      end
    end
```
In this case, overriding the `nickname` doesn't work and I get the auto-generated one. Any idea? I'm using `grape-swagger` 0.22.

### PR DESCRIPTION
**Pull request by pbendersky**
adds support for nicknamed routes (fixes #481)

### CODE REVIEW

File: spec/swagger_v2/nicknamed_api_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+describe 'a nicknamed mounted api' do
+  before :all do
```
Comment (Bugagazavr) [inline spec/swagger_v2/nicknamed_api_spec.rb]: please use `let`

File: spec/swagger_v2/nicknamed_api_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    class NicknamedMountedApi < Grape::API
```
Comment (Bugagazavr) [inline spec/swagger_v2/nicknamed_api_spec.rb]: also should be `Class.new(Grape::API) do`, thanks

File: spec/swagger_v2/nicknamed_api_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe 'a nicknamed mounted api' do
+  def app
```
Comment (Bugagazavr) [inline spec/swagger_v2/nicknamed_api_spec.rb]: just use `let!(:app) do` instead method definition

**Comment by Bugagazavr** [comment]
Thanks, do not forget fix rubocop errors, you can fix this automatically via `rubocop -a`

/cc @LeFnord

**Comment by pbendersky** [comment]
Done, @Bugagazavr. Thank you for the quick feedback.

**Comment by LeFnord** [comment]
:+1: … I really like it … filling the skeleton with functionality

**Comment by Bugagazavr** [comment]
@pbendersky thanks, now you need add this feature to `CHANGELOG.md`

**Comment by pbendersky** [comment]
@Bugagazavr fix or feature? I'm leaning for fix, as it's in the README, but I can see it as a feature as well.

**Comment by Bugagazavr** [comment]
@pbendersky you can add this as fix.

**Comment by pbendersky** [comment]
All set then!

**Comment by Bugagazavr** [comment]
Great job, thanks again

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@
 * [#476](https://github.com/ruby-grape/grape-swagger/pull/476): Fixes for handling the parameter type when body parameters are defined inside desc block - [@anakinj](https://github.com/anakinj).
 * [#478](https://github.com/ruby-grape/grape-swagger/pull/478): Refactors building of properties, corrects documentation of array items - [@LeFnord](https://github.com/LeFnord).
 * [#479](https://github.com/ruby-grape/grape-swagger/pull/479): Fix regex for Array and Multi Type in doc_methods. Parsing of "[APoint]" should return "APoint" - [@frodrigo](https://github.com/frodrigo).
+* [#483](https://github.com/ruby-grape/grape-swagger/pull/483): Added support for nicknamed routes - [@pbendersky](https://github.com/pbendersky)
```
Comment (dblock) [inline CHANGELOG.md]: This is missing a period ;)

Comment (pbendersky) [inline CHANGELOG.md]: Added in #485

**Pull request merged**